cpp/tests/datetime/datetime_ops_test.cu

```diff
@@ -0,0 +1,154 @@
+  expect_columns_equal(*cudf::datetime::extract_year(test_timestamps_ms_view), fixed_width_column_wrapper<int16_t>{1965, 2018, 2023});
+
+  expect_columns_equal(*cudf::datetime::extract_month(test_timestamps_D_view), fixed_width_column_wrapper<int16_t>{10, 7, 1});
+  expect_columns_equal(*cudf::datetime::extract_month(test_timestamps_s_view), fixed_width_column_wrapper<int16_t>{10, 7, 1});
+  expect_columns_equal(*cudf::datetime::extract_month(test_timestamps_ms_view), fixed_width_column_wrapper<int16_t>{10, 7, 1});
+
+  expect_columns_equal(*cudf::datetime::extract_day(test_timestamps_D_view), fixed_width_column_wrapper<int16_t>{26, 4, 25});
+  expect_columns_equal(*cudf::datetime::extract_day(test_timestamps_s_view), fixed_width_column_wrapper<int16_t>{26, 4, 25});
+  expect_columns_equal(*cudf::datetime::extract_day(test_timestamps_ms_view), fixed_width_column_wrapper<int16_t>{26, 4, 25});
+
+  expect_columns_equal(*cudf::datetime::extract_weekday(test_timestamps_D_view), fixed_width_column_wrapper<int16_t>{2, 3, 3});
+  expect_columns_equal(*cudf::datetime::extract_weekday(test_timestamps_s_view), fixed_width_column_wrapper<int16_t>{2, 3, 3});
+  expect_columns_equal(*cudf::datetime::extract_weekday(test_timestamps_ms_view), fixed_width_column_wrapper<int16_t>{2, 3, 3});
+
+  expect_columns_equal(*cudf::datetime::extract_hour(test_timestamps_D_view), fixed_width_column_wrapper<int16_t>{0, 0, 0});
+  expect_columns_equal(*cudf::datetime::extract_hour(test_timestamps_s_view), fixed_width_column_wrapper<int16_t>{14, 12, 7});
+  expect_columns_equal(*cudf::datetime::extract_hour(test_timestamps_ms_view), fixed_width_column_wrapper<int16_t>{14, 12, 7});
+
+  expect_columns_equal(*cudf::datetime::extract_minute(test_timestamps_D_view), fixed_width_column_wrapper<int16_t>{0, 0, 0});
+  expect_columns_equal(*cudf::datetime::extract_minute(test_timestamps_s_view), fixed_width_column_wrapper<int16_t>{1, 0, 32});
+  expect_columns_equal(*cudf::datetime::extract_minute(test_timestamps_ms_view), fixed_width_column_wrapper<int16_t>{1, 0, 32});
+
+  expect_columns_equal(*cudf::datetime::extract_second(test_timestamps_D_view), fixed_width_column_wrapper<int16_t>{0, 0, 0});
+  expect_columns_equal(*cudf::datetime::extract_second(test_timestamps_s_view), fixed_width_column_wrapper<int16_t>{12, 0, 12});
+  expect_columns_equal(*cudf::datetime::extract_second(test_timestamps_ms_view), fixed_width_column_wrapper<int16_t>{12, 0, 12});
+
+}
+
+TYPED_TEST(DatetimeOpsTest, TestExtractingGeneratedDatetimeComponents) {
+
+  using namespace cudf::test;
+  using namespace simt::std::chrono;
+  using Rep = typename TypeParam::rep;
+  using Period = typename TypeParam::period;
+
+  auto start = milliseconds(-2500000000000); // Sat, 11 Oct 1890 19:33:20 GMT
+  auto stop_ = milliseconds( 2500000000000); // Mon, 22 Mar 2049 04:26:40 GMT
+  auto test_timestamps = generate_timestamps<Rep, Period>(this->size(),
```

Comment:
`generate_timestamps` looks like it should be modified to use `fixed_width_column_wrapper`, or just instead provide iterators you can feed into `fixed_width_column_wrapper` that do the same thing.

Reply:
I opted to keep the `generate_timestamps` utility function but change its signature and implementation to return a `fixed_width_column_wrapper`. It's basically [`numpy.arange()`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.arange.html) or [`pd.date_range`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.date_range.html) specialized for timestamps, so it seems useful to have a utility function rather than duplicating this logic in each test.